ambry-cloud/src/main/java/com/github/ambry/cloud/CloudBlobStore.java

```diff
@@ -538,7 +538,7 @@ private void validateAccountAndContainer(Map<String, CloudBlobMetadata> cloudBlo
    */
   private boolean preDeleteValidation(CloudBlobMetadata metadata, StoreKey key, Map<String, Object> updateFields)
       throws StoreException {
-    validateAccountAndContainer(Collections.singletonMap(metadata.getId(), metadata), Collections.singletonList(key));
+    validateAccountAndContainer(Collections.singletonMap(key.getID(), metadata), Collections.singletonList(key));
```

Comment:
Do we also need to change the `metadata.getId()` at line 554?

Reply:
fixed.